cmd/queue/main.go

```diff
@@ -126,6 +126,25 @@ func isProbe(r *http.Request) bool {
 	return strings.HasPrefix(r.Header.Get("User-Agent"), "kube-probe/")
 }
 
+func timeoutHandler(h http.Handler, d time.Duration) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		done := make(chan bool)
```

Comment:
Can this be a `done := make(chan struct{})` which is `close`d by the goroutine? Just like in the original implementation.

Reply:
Yes, good idea!